pytorch_lightning/core/step_result.py

```diff
@@ -367,7 +367,10 @@ def get_forked_metrics(self, add_dataloader_idx=False):
             dl_key = self._add_dataloader_idx(k, options["dataloader_idx"], add_dataloader_idx)
 
             if options['forked']:
-                result[dl_key] = self[k]
+                if isinstance(self[k], Metric):
+                    result[dl_key] = self[k].compute().detach()
```

Comment:
I cannot find the `detach()` in the new class? Is it not needed anymore?

Reply:
Good catch ! I forgot it.